R/parse_expr.R

```diff
@@ -133,7 +133,7 @@ parse_expr_assignment_lhs <- function(lhs, src, call) {
     update = function(name) NULL,
     deriv = function(name) NULL,
     output = function(name) NULL,
-    dim = function(name) NULL,
+    dim = function(name, ...) NULL,
```

Comment:
The code just below catches the no-arg error `dim() <- ` - so I was thinking of doing it as above to ensure I get at least one arg. 

Then the idea is that `name` is a bit different in that it's the only one that might be a real dim; the others in `...` will all be aliases. (`name` might also be an alias if there's a dim() on the RHS, but only `name` can ever be a real dim)

Reply:
Whether I really make use of that distinction though...

Reply:
It's easy to test no args though - that's just checking the length. And because we the first argument could be an alias I don't think this distinction has an upside, just more logic to cope with